src/vm/analysis/type_checker/natives/mod.rs

```diff
@@ -300,6 +300,31 @@ impl TypedNativeFunction {
                          TypeSignature::UIntType,
                          TypeSignature::IntType],
                     BUFF_32.clone()))),
+            StxTransfer =>
+                Simple(SimpleNativeFunction(FunctionType::Fixed(FixedFunction {
+                    args: vec![
+                        FunctionArg::new(TypeSignature::UIntType,
+                                         ClarityName::try_from("amount".to_owned())
+                                         .expect("FAIL: ClarityName failed to accept default arg name")),
+                        FunctionArg::new(TypeSignature::PrincipalType,
```

Comment:
Wait, doesn't this mean that anyone can transfer STX from anyone to anyone else?  Shouldn't this only take a recipient principal, and have the sender be automatically tx-sender?

Reply:
Ah, I see elsewhere that the sender must be `tx-sender`.  But the question remains, why have this argument at all?  Is it meant to be a sanity check?

Reply:
Yep -- the idea is to use the argument as a sanity check.